pkg/apiserver/registry/networkpolicy/addressgroup/rest.go

```diff
@@ -71,12 +72,20 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 }
 
 func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	labelSelector := labels.Everything()
+	if options != nil && options.LabelSelector != nil {
+		labelSelector = options.LabelSelector
+	}
 	addressGroups := r.addressGroupStore.List()
-	list := new(controlplane.AddressGroupList)
-	list.Items = make([]controlplane.AddressGroup, len(addressGroups))
+	items := make([]controlplane.AddressGroup, 0, len(addressGroups))
 	for i := range addressGroups {
-		store.ToAddressGroupMsg(addressGroups[i].(*types.AddressGroup), &list.Items[i], true)
+		var item controlplane.AddressGroup
+		store.ToAddressGroupMsg(addressGroups[i].(*types.AddressGroup), &item, true)
+		if labelSelector.Matches(labels.Set(item.Labels)) {
```

Comment:
Could we have labels on addressGroups?

Reply:
When LabelSelector is nil, does it deserve a special handling to skip labelSelector.Matches?

Reply:
labelSelector is `Everything` when it's not set, and it's actually an empty map, the `Matches` function actually returns true directly, by special handling, you mean one path to not calling labelSelector.Matches at all and another does? could it be redundant?
The performance difference is very small even listing 10,000 objects.
```
func filter1(addressGroups []*types.AddressGroup, labelSelector labels.Selector) []controlplane.AddressGroup{
	items := make([]controlplane.AddressGroup, 0, len(addressGroups))
	for i := range addressGroups {
		var item controlplane.AddressGroup
		store.ToAddressGroupMsg(addressGroups[i], &item, true)
		if labelSelector.Matches(labels.Set(item.Labels)) {
			items = append(items, item)
		}
	}
	return items
}

func filter2(addressGroups []*types.AddressGroup, labelSelector labels.Selector) []controlplane.AddressGroup{
	items := make([]controlplane.AddressGroup, 0, len(addressGroups))
	if labelSelector.Empty() {
		for i := range addressGroups {
			var item controlplane.AddressGroup
			store.ToAddressGroupMsg(addressGroups[i], &item, true)
			items = append(items, item)
		}
	} else {
		for i := range addressGroups {
			var item controlplane.AddressGroup
			store.ToAddressGroupMsg(addressGroups[i], &item, true)
			if labelSelector.Matches(labels.Set(item.Labels)) {
				items = append(items, item)
			}
		}
	}
	return items
}
func benchmarkFilter(b *testing.B, filter func(addressGroups []*types.AddressGroup, labelSelector labels.Selector) []controlplane.AddressGroup) {
	addressGroups := []*types.AddressGroup{}
	for i:=0;i<10000;i++{
		addressGroups = append(addressGroups, &types.AddressGroup{
			Name: fmt.Sprintf("%d", i),
		})
	}
	b.ReportAllocs()
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		filter(addressGroups, labels.Everything())
	}
}

func BenchmarkFilter1(b *testing.B) {
	benchmarkFilter(b, filter1)
}

func BenchmarkFilter2(b *testing.B) {
	benchmarkFilter(b, filter2)
}
```

Got:
```
BenchmarkFilter1
BenchmarkFilter1-4   	     688	   1595190 ns/op	 3285025 B/op	       2 allocs/op
BenchmarkFilter2
BenchmarkFilter2-4   	     706	   1669110 ns/op	 3285028 B/op	       2 allocs/op
```

Reply:
Ok. Then it might not be worthwhile.